### DISCUSSION TITLE
Distributing the S-Function for Matlab

### DISCUSSION THREAD

**Discussion by bcoconni** [JSBSim-Team/jsbsim#484] [Ideas]
Since I am quite unfamiliar with the Matlab community, could someone enlighten me about the usual way of distributing open source S-Functions to the Matlab community ?

Now that the PR #445 is merged, we are providing the source code to build the S-Function for JSBSim which is great. However, at the moment a Matlab user that would want to use JSBSim S-Function needs to build everything from scratch using either CMake or Visual Studio. Not sure if this is very appealing to the average Matlab user ? Is there a means to distribute JSBSim S-Function that would reduce the burden of a Matlab user ?

While I browsed the internet, I saw that distributing compiled `mex` files seems not to be the way to go since you need to get an S-function which has been compiled for the very version of Matlab that you are using, under the same platform and most likely with the same compiler.

Any opinion ?

**Comment by agodemar** [comment]
@bcoconni I think that as we begin to support Matlab S-Functions in JSBSim, it's fine having the user to take care of mex file builds.

As a matter of fact, the visual aids that Matlab offers nowadays to create apps and mex files are pretty advanced and streamlined. This is called Matlab *deploytool*.

The work on the JSBSim side should be focused on having the best possible interface to Matlab, one which is general and simple enough to provide the basis for more complex projects. And I hope this will encourage more users to contribute.

**Comment by bcoconni** [comment]
Thanks for your answer @agodemar.

I have modified our Windows installer to install the files needed to build the Matlab S-Function. Note that the item `JSBSim files to build an interface to Matlab` is a subitem of `JSBSim development files (headers and C++ library)`. This means that as soon as the subitem is ticked then its parent item will also be ticked. This is to make sure that all the files needed to build the S-Function from Matlab are installed, including the headers and the static library `JSBSim.lib`.

![JSBSim installer](https://user-images.githubusercontent.com/7502904/126895161-1939dd1b-08aa-44c6-86c9-e231d2612b61.PNG)

Once the files are installed, the folder structure is as displayed below. 

![JSBSim installer tree](https://user-images.githubusercontent.com/7502904/126895412-1e3bb36c-984f-4a4f-9f7c-51d28de7e00c.PNG)

I guess the file `matlab/JSBSimSimulinkCompile.m` should be modified to match this directory structure ? @agodemar, @sthelia any opinion ?
https://github.com/JSBSim-Team/jsbsim/blob/4aaf5e52f7454c2f9aa43d27c2c326d1c8b8e3b6/matlab/JSBSimSimulinkCompile.m#L1-L10

**Reply by sthelia** [reply]
> I guess the file `matlab/JSBSimSimulinkCompile.m` should be modified to match this directory structure ? @agodemar, @sthelia any opinion ?
> 
> https://github.com/JSBSim-Team/jsbsim/blob/4aaf5e52f7454c2f9aa43d27c2c326d1c8b8e3b6/matlab/JSBSimSimulinkCompile.m#L1-L10

Yes, I believe modifying the ` mex ` command would be enough to allow a matlab user to easily be able to run JSBSim!
